src/p4p/test/system_client_server.py

```diff
@@ -0,0 +1,631 @@
+    #   pvget TESTCOUNTER -r junk.thing
+    def testGetBadSubfield(self):
+        if self.MALCOLM:
+            # Currently only returns an error structure as pvaPy can't raise
+            # exceptions
+            error = self.ctxt.get("TESTCOUNTER", "junk.thing")
+            self.assertEqual(error.getID(), "malcolm:core/Error:1.0")
+            self.assertEqual(error.message, "UnexpectedError: Object of type 'malcolm:core/Block:1.0' has no attribute 'junk'")
+        else:
+            # TODO: What is the right error to get here?
+            with self.assertRaises(KeyError):
+                self.ctxt.get("TESTCOUNTER", "junk.thing")
+
+    # Equivalent to:
+    #   pvget BADCHANNEL -r ""
+    def testGetBadChannel(self):
+        with self.assertRaises(TimeoutError):
+            self.ctxt.get("BADCHANNEL", timeout=1.0)
+
+    # Equivalent to:
+    #   pvget TESTCOUNTER.meta -r fields
+    def testGetDottedSubfield(self):
+        meta = self.ctxt.get("TESTCOUNTER.meta", "fields")
+        self.assertEqual(meta.getID(), "structure")
+        self.assertEqual(len(meta.items()), 1)
+        self.assertEqual(len(meta.fields), 4)
+        fields_code = dict(meta.type().aspy()[2])["fields"]
+        self.assertEqual(fields_code, "as")
+
+    # Equivalent to:
+    #   pvget TESTCOUNTER.counter -r ""
+    def testGetDottedNTScalar(self):
+        # Use the one with unwrapping on
+        ctxt = Context("pva")
+        self.addCleanup(ctxt.close)
+        counter = ctxt.get("TESTCOUNTER.counter")
+        self.assertIsInstance(counter, ntfloat)
+        self.assertEqual(counter, 0.0)
+        self.assertEqual(counter.severity, 0)
+        self.assertEqual(counter.raw.getID(), "epics:nt/NTScalar:1.0")
+
+    # Equivalent to:
+    #   pvget TESTCOUNTER.counter -r ""
+    def testGetDottedNTScalarNoUnwrap(self):
+        counter = self.ctxt.get("TESTCOUNTER.counter")
+        self.assertEqual(counter.getID(), "epics:nt/NTScalar:1.0")
+        self.assertEqual(counter.value, 0.0)
+        self.assertEqual(counter.timeStamp.getID(), "time_t")
+        self.assertEqual(counter.alarm.getID(), "alarm_t")
+        self.assertEqual(counter.alarm.severity, 0)
+
+    def assertCounter(self, value):
+        if self.MALCOLM:
+            counter = self.ctxt.get("TESTCOUNTER.counter").value
+        else:
+            counter = self.counter_structure.counter.value
+        self.assertEqual(counter, value)
+
+    # Should be equivalent to (except pvput doesn't do the right thing):
+    #   pvput TESTCOUNTER -r "counter.value" 5
+    def testPut(self):
+        self.assertCounter(0)
+        self.ctxt.put("TESTCOUNTER", 5, "counter.value")
+        self.assertCounter(5)
+        self.ctxt.put("TESTCOUNTER", 0, "counter.value")
+        self.assertCounter(0)
+
+    # Equivalent to:
+    #   pvput TESTCOUNTER.counter 5
+    def testPutDotted(self):
+        self.assertCounter(0)
+        # TODO: pvput defaults the request to "value", should p4p do the same?
```

Comment:
No!  Early on, when P4P had only an RPC server, I was helping a junior programmer learn about PVA clients via. pvaPy and via. email.  There was a whole afternoon lost answering the question "where is the timeStamp" in part because of this default.  This left a mark on me.  I'm firmly in favor of novice friendly defaults, even when this impacts performance.  So the default will be everything.  Interested users can then use pvRequest to select a sub-set of everything.

Reply:
Good, I discussed this with @Araneidae and he came to the same conclusion